neuro_san/session/abstract_http_service_agent_session.py

```diff
@@ -0,0 +1,127 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import Dict
+
+from leaf_common.time.timeout import Timeout
+
+from neuro_san.interfaces.agent_session_constants import AgentSessionConstants
+
+
+class AbstractHttpServiceAgentSession(AgentSessionConstants):
+    """
+    Abstract base class for sync and async version for http service agent sessions.
+    """
```

Comment:
The members and methods in this class are common to AsyncHttpServiceAgentSession and HttpServiceAgentSession.  Consolidate for a single source of truth.